### ISSUE TITLE
PayPal Plus / Pro Hosted Integration

### ISSUE THREAD

**Issue by angelleye** [angelleye/paypal-woocommerce#188]
https://www.paypal.com/de/webapps/mpp/paypal-plus

**Comment by levansy** [comment]
You have any document?

**Comment by angelleye** [comment]
No need to worry about this one yet.

**Comment by angelleye** [comment]
This is essentially the same thing as PayPal Advanced except that it's for other countries.

https://www.paypalobjects.com/webstatic/en_JP/developer/docs/pdf/paymentsplus.pdf

**Comment by angelleye** [comment]
PayPal actually calls this different things depending on the country.  We should review this info as well as the PDF above:

https://developer.paypal.com/webapps/developer/docs/classic/products/website-payments-pro-hosted-solution/

**Comment by angelleye** [comment]
[Here is another document that covers PayPal Plus on the REST API](http://fiber.angelleye.com/forums/github/paypal-plus-rest-api/), which should be compatible with German PayPal accounts (unlike the previous documents).

We need to focus on getting this one done first of all, and then if the others aren't included in this we can see about getting those done, too.  We may move those to a later update, though.  Right now the German version is the most important.

**Comment by nishitlangaliya** [comment]
Hi Andrew,

I have pushed paypal plus code to Issue-188C branch for a review . Please check for the same when you get a chance and let me know your thoughts and suggestions for the same.

**Comment by nishitlangaliya** [comment]
Removed conflicts  and Pushed to 'Issue-188G' branch and also merged to 'dev' branch for a review.

**Comment by angelleye** [comment]
I found another piece of documentation here regarding PayPal Plus:  https://www.paypalobjects.com/webstatic/de_DE/downloads/PayPal-PLUS-IntegrationGuide.pdf

Just want to verify that the flow they show starting on page 26 is what we're getting here.

**Comment by levansy2020** [comment]
Please check and give feedback

**Comment by angelleye** [comment]
Looks like you removed much of what you had originally put in place..??  Were you able to fully test orders using the sandbox?

**Comment by levansy2020** [comment]
Please check Andrew, branch release-merge

**Comment by angelleye** [comment]
I pulled latest changes and I'm still having some trouble here. 

 * I am signed in during testing.
 * I clicked Place Order for PayPal Plus and was sent to PayPal, however, instead of completing the order I clicked the Cancel link and was returned back to the cart page on the site as expected.
 * From that point on, when I go back to the checkout page and try to Place Order again with PayPal Plus it just sends me to the home page with this URL:  http://woo.angelleye.com/?v=fef5f5d4a1dd

**Comment by angelleye** [comment]
I cleared my cookies for the site, which logged me out, and then I tried again.  This time the Place Order button did work again, and once again I clicked the cancel link.  

I was wrong before, though.  It's not sending us to the cart page when I do that, it's just going to the home page with that same URL:  http://woo.angelleye.com/?v=fef5f5d4a1dd

Once again, after doing that, I cannot try again.  Clicking Place order just keeps sending me back to that same URL instead of moving forward to PayPal again.

So we need to get that worked out, and then we also need to add the option to set what the cancel URL should be just like we have for Express Checkout.
